zerver/tests/test_decorators.py

```diff
@@ -108,6 +113,92 @@ def test(request, payload=REQ(argument_type="invalid")):
                 pass
             test(request)
 
+    def test_api_key_only_webhook_view(self):
+        @api_key_only_webhook_view('ClientName')
+        def get_user_profile_api_key(request, user_profile, client):
+            return user_profile.api_key
+
+        class Request(object):
+            REQUEST = {} # type: Dict[str, str]
+            COOKIES = {}
+            META = {'PATH_INFO': ''}
+
+        webhook_bot_email = 'webhook-bot@zulip.com'
+        request = Request()
+
+        request.REQUEST['api_key'] = 'not_existing_api_key'
+        with self.assertRaises(JsonableError):
+            get_user_profile_api_key(request)
+
+        request.REQUEST['api_key'] = get_user_profile_by_email(webhook_bot_email).api_key
+        request.COOKIES['csrftoken'] = get_token(request)
+        self.assertEqual(get_user_profile_api_key(request), get_user_profile_by_email(webhook_bot_email).api_key)
+
+    def test_authenticated_json_post_view(self):
+        @authenticated_json_post_view
+        def get_user_profile_api_key(request, user_profile, client):
+            return user_profile.api_key
+
+        webhook_bot_email = 'webhook-bot@zulip.com'
+        user = get_user_profile_by_email(webhook_bot_email)
+
+        class Request(object):
+            REQUEST = {} # type: Dict[str, str]
+            COOKIES = {}
+            META = {'PATH_INFO': ''}
+            method = 'POST'
+
+        request = Request()
+        request.user = user
```

Comment:
Ah,, well there's two solutions here:
* One is that that you can use a test urlconf: https://docs.djangoproject.com/en/1.9/topics/testing/tools/#urlconf-configuration
* The second is that you can just pick a route that uses that auth decorator, and call it.

Also, looking at these now, I think it'd improve these tests verified that the JsonableError that is raised actually is the expected one (via checking the error message string in the JsonableError).  Since this is a common pattern, maybe it makes sense to add a new test helper function that's `self.assertJsonableError("error text")` and use that for these tests...